daft/dataframe/dataframe.py

```diff
@@ -158,6 +169,45 @@ def _result(self) -> Optional[PartitionSet]:
         else:
             return self._result_cache.value
 
+    def _explain_broadcast(self):
+        import json
+        from urllib import request
+
+        from daft.dataframe.display import MermaidFormatter
+
+        try:
+            enable_dashboard = int(os.environ.get("DAFT_DASHBOARD_ENABLE", "0"))
+        except ValueError:
+            return
+
+        if not enable_dashboard:
+            return
+
+        addr = os.environ.get("DAFT_DASHBOARD_ADDR", "localhost")
+        port = int(os.environ.get("DAFT_DASHBOARD_PORT", 3238))
+
+        is_cached = self._result_cache is not None
+        plan_time_start = datetime.now(timezone.utc)
+        mermaid_plan = MermaidFormatter(
+            builder=self.__builder, show_all=True, simple=False, is_cached=is_cached
+        )._repr_markdown_()
+        plan_time_end = datetime.now(timezone.utc)
+
+        headers = {
+            "Content-Type": "application/json",
+        }
+        data = json.dumps(
+            {
+                "id": str(uuid4()),
```

Comment:
This should ideally be generated server-side

Reply:
(non-blocking)

Reply:
Will punt this until later.